lib/Runtime/Language/InlineCache.cpp

```diff
@@ -26,8 +26,6 @@ namespace Js
         // We may, however, have a flags cache (setter) change to add property cache.
         Assert(typeWithoutProperty == nullptr || !IsProto());
 
-        requestContext->RegisterAsScriptContextWithInlineCaches();
```

Comment:
no, it was not setting the flag before, it only check and allocate the item in the zero list, and save the item address back to scriptContext. the flag was set in ScriptContext::RegisterProtoInlineCache and ScriptContext::RegisterStoreFieldInlineCache, which is the call right after